Sources/Public/CalendarViewContent.swift

```diff
@@ -208,19 +175,27 @@ public final class CalendarViewContent {
   /// displayed. The `CalendarItemModel`s that you return will be used to create the views for each month header in
   /// `CalendarView`.
   ///
-  /// If you don't configure your own month header item provider via this function, then a default month header item provider will be
-  /// used.
+  /// If you don't configure your own month header item provider via this function, or if the `monthHeaderItemProvider` closure
+  /// returns nil, then a default month header item provider will be used.
   ///
   /// - Parameters:
   ///   - monthHeaderItemProvider: A closure (that is retained) that returns a `CalendarItemModel` representing a
   ///   month header.
   ///   - month: The `Month` for which to provide a month header item.
   /// - Returns: A mutated `CalendarViewContent` instance with a new month header item provider.
   public func monthHeaderItemProvider(
-    _ monthHeaderItemProvider: @escaping (_ month: Month) -> AnyCalendarItemModel)
+    _ monthHeaderItemProvider: @escaping (_ month: Month) -> AnyCalendarItemModel?)
     -> CalendarViewContent
   {
-    self.monthHeaderItemProvider = monthHeaderItemProvider
+    self.monthHeaderItemProvider = { [defaultMonthHeaderItemProvider] month in
+      guard let itemModel = monthHeaderItemProvider(month) else {
+        // If the caller returned nil, fall back to the default item provider
+        return defaultMonthHeaderItemProvider(month)
```

Comment:
To avoid having to change any of the existing call sites, we can have the instance properties still always return a non-nil value, even if the user-provided closure returns nil. An easy way to do this is to just fall back to the existing default item provider, which always returns a valid value.